Sources/ArcGISToolkit/Components/Offline/OfflineMapViewModel.swift

```diff
@@ -126,9 +135,12 @@ class OfflineMapViewModel: ObservableObject {
         // undetermined.
         if mode == .undetermined || mode == .onDemand {
             await loadOnDemandMapModels()
-            // If there are any on-demand areas at all, and the mode is
-            // undetermined, then set mode to on-demand.
-            if mode == .undetermined, !onDemandMapModels.isEmpty {
+            // If the mode is undetermined, then set mode to on-demand.
+            // If there are on-demand map areas, the mode should be set to
+            // on-demand. If there are no on-demand map areas, we can assume
+            // that the user started with an empty map and want to add on-demand
+            // map areas.
+            if mode == .undetermined {
```

Comment:
When there is no Internet connection, the method returns early with `mode == .undetermined`. If the method reaches this part, it means:

- network is connected
- mode is not preplanned, is undetermined

In this case, I think it is safe to assume the mode should be onDemand.

Reply:
I applied Ryan's suggestion to add the `noInternetAvailable` and renamed `undetermined` to `ambiguous`